### PR TITLE
Add setup steps for pre-sandbox dependency installation

### PR DESCRIPTION
**Pull request by Gkrumbach07**
## Summary
- Adds a new `setup` key to `.agentic-ci/config.yml` that defines commands to run on the host before uploading the workdir into the OpenShell sandbox
- Setup steps execute with full internet access, enabling dependency installation (e.g. `npm ci`) whose outputs are then available inside the sandbox
- Supports both GitHub Actions-style objects (`name`/`run`) and bare-string shorthand

## Motivation

Repos with heavy dependency installs (e.g. ODH dashboard needing `npm install`) fail in autofix because the OpenShell sandbox blocks internet access. This feature runs setup commands outside the sandbox so their outputs (e.g. `node_modules/`) are included in the workdir upload.

Slack thread: https://redhat-internal.slack.com/archives/C0ASJ32PJ0N/p1783524952468199
Resolves: [RHAIFIRST-267](https://redhat.atlassian.net/browse/RHAIFIRST-267)

## Config format

```yaml
# .agentic-ci/config.yml
setup:
  - name: Install dependencies
    run: npm ci
  - npm run build  # bare string shorthand
```

## Test plan
- [x] Unit tests for config parsing (10 tests covering all forms, edge cases)
- [x] Full test suite passes (737 tests)
- [x] Ruff lint clean
- [x] Ruff format clean
- [x] Mypy typecheck clean
- [ ] E2E test with ODH dashboard repo

🤖 Generated with [Claude Code](https://claude.com/claude-code)

[RHAIFIRST-267]: https://redhat.atlassian.net/browse/RHAIFIRST-267?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Summary by CodeRabbit

- **New Features**
  - Added repository configuration loading from `.agentic-ci/config.yml` (optional `setup`).
  - OpenShell now executes configured host-side setup steps before uploading the work directory to the sandbox, so setup outputs are available in the run (fail-fast on errors with per-step time limits).
- **Documentation**
  - Added/expanded guidance on `setup` steps execution flow and OpenShell network/policy configuration references.
- **Tests**
  - Added unit coverage for missing/malformed configs and for skipping invalid `setup` entries.
  - Added an end-to-end OpenShell sandbox test confirming setup steps run on the host before upload.

### CODE REVIEW

**Comment by EmilienM** [comment]
@Gkrumbach07 we need docs and agents.md update please

File: docs/backends/openshell.md [hunk 1]
```diff
@@ -176,6 +178,21 @@ openshell gateway remove ci                      # deregister from CLI
 # Gateway and podman service processes are killed by PID
 ```
 
+## Setup Steps
+
+Because the sandbox has no internet access by default, repositories that
```
Comment (EmilienM) [inline docs/backends/openshell.md:183]: it is wrong to say that, it has a limited access to the allowed-list hosts (like github.com, etc). I'd rephrase it to avoid confusion.

File: docs/configuration.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
+# Project Configuration
+
```
Comment (EmilienM) [inline docs/configuration.md:2]: If you are separating openshell backend configuration (which is fine btw!) - you could have also a section for the `openshell-policy.yml`, and maybe trim it from `openshell.md` to avoid content duplication?

In this commit, you have a well documented `openshell-policy.yml` in `openshell.md` and a whell documented `.agentic-ci/config.yml` in the new `configuration.md`, it's confusing.

File: src/agentic_ci/backends/openshell/__init__.py [hunk 3]
```diff
@@ -143,6 +146,22 @@ def _upload_sandbox_config(self, otel_enabled=False):
         finally:
             shutil.rmtree(config_dir, ignore_errors=True)
 
+    def _run_setup_steps(self):
```
Comment (EmilienM) [inline src/agentic_ci/backends/openshell/__init__.py]: I'd like to see e2e tests for that feature, for all backends: `local`, `podman` and `openshell`.
Not tested, not working ;-)

**Comment by Gkrumbach07** [comment]
Reopening from an upstream branch so e2e CI jobs can run. See new PR.
